src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/CreateSdkSymbolsLayout.cs

```diff
@@ -0,0 +1,161 @@
+        {
+            List<string> filesWithoutPDBs = new List<string>();
+
+            if (Directory.Exists(SdkSymbolsLayoutPath))
+            {
+                Directory.Delete(SdkSymbolsLayoutPath, true);
+            }
+
+            foreach (string file in Directory.GetFiles(SdkLayoutPath, "*", SearchOption.AllDirectories))
+            {
+                if (file.EndsWith(".dll", StringComparison.InvariantCultureIgnoreCase) &&
+                    !file.EndsWith(".resources.dll", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    string guid = string.Empty;
+                    using var pdbStream = File.OpenRead(file);
+                    using var peReader = new PEReader(pdbStream);
+                    try
+                    {
+                        // Check if pdb is embedded
+                        if (peReader.ReadDebugDirectory().Any(entry => entry.Type == DebugDirectoryEntryType.EmbeddedPortablePdb))
+                        {
+                            continue;
+                        }
+
+                        var debugDirectory = peReader.ReadDebugDirectory().First(entry => entry.Type == DebugDirectoryEntryType.CodeView);
+                        var codeViewData = peReader.ReadCodeViewDebugDirectoryData(debugDirectory);
+                        guid = $"{codeViewData.Guid.ToString("N").Replace("-", string.Empty)}";
+                    }
+                    catch (Exception)
+                    {
+                        // Ignore binaries without debug info
+                        continue;
+                    }
+
+                    if (guid != string.Empty)
+                    {
+                        if (!allPdbGuids.ContainsKey(guid))
+                        {
+                            filesWithoutPDBs.Add(file.Substring(SdkLayoutPath.Length + 1));
+                        }
+                        else
+                        {
+                            // Copy matching pdb to symbols path, preserving sdk binary's hierarchy
+                            string sourcePath = (string)allPdbGuids[guid]!;
+                            string destinationPath =
+                                file.Replace(SdkLayoutPath, SdkSymbolsLayoutPath)
+                                    .Replace(Path.GetFileName(file), Path.GetFileName(sourcePath));
+
+                            Directory.CreateDirectory(Path.GetDirectoryName(destinationPath)!);
+                            File.Copy(sourcePath, destinationPath, true);
+                        }
+                    }
+                }
+            }
+
+            return filesWithoutPDBs;
+        }
+
+        public Hashtable IndexAllSymbols()
+        {
+            Hashtable allPdbGuids = new Hashtable();
+
+            foreach (string file in Directory.GetFiles(AllSymbolsPath, "*.pdb", SearchOption.AllDirectories))
+            {
+                try
```

Comment:
Try-catch block isn't needed here - will remove it.

Reply:
Fixed in https://github.com/dotnet/installer/pull/17454/commits/afd434a09fe8b8a68a598741a98df9304259865a